### ISSUE TITLE
NaNoGenMo 2016: Where to host it?

### ISSUE THREAD

**Issue by dariusk** [dariusk/NaNoGenMo-2015#199]
For the last few years I've hosted NaNoGenMo on a Github repository like this one. I've considered it a decent choice because lots of coders already have a Github account, and a Github issue is basically a forum thread with really nice syntax highlighting.

However, there might be better places to organize the 2016 event! Does anyone have any thoughts on where might be better and why?

**Comment by greg-kennedy** [comment]
Github works for me, you may also consider making a subreddit with fancy flairs for COMPLETED, IN PROGRESS, etc.

**Comment by enkiv2** [comment]
Github makes a lot of sense, particularly for everybody who has
participated before, but the way we use the issue system has always been
confusing for people who are new to it. I'm not opposed to a subreddit if
this is deemed likely to lower confusion.
Personally, I respond to issues via email & treat this like a newsgroup; if
this is more common than interaction in the Github website itself, then
something like Google Groups might be an option too -- similar threaded
conversation interface with email integration, but without tags, however on
the other hand the conversation threads are clearly conversation threads
rather than leading people to believe they are bug reports.

**Comment by liambaloh** [comment]
If you want to host it yourself, you might want to consider https://github.com/OneHourGameJam/OneHourGameJam as a software solution to power it. If you're interested, let me know, so I have enough time to develop a few extensions (for example hosting text). 

For a live demo, see http://onehourgamejam.com/

**Comment by hugovk** [comment]
GitHub works for me.

There's also things like [hackathon.io](http://www.hackathon.io/events) and [WeHack.It](http://www.wehack.it/hackathons/active#welcome-hackathons) but I've never used them.

**Comment by ikarth** [comment]
Pros for github:
- Consistency (we've always done it this way!)
- Historical continuity
- Archiving: this is a big one: we've got a reliable record of the entire history of NaNoGenMo, stored indefinitely
- Having it on GitHub makes source code delivery obvious and open-source. I worry that we'd lose some of the full participation in releasing sources if we changed away from GitHub. Not that people wouldn't want to, but they just wouldn't get around to uploading it...

Cons for GitHub:
- Having the discussions in the issues tends to confuse new people (maybe add an explanation in the readme?)
- Every year, there are people who post in the previous year's project
- Would another location have more visibility/be easier for non-programmers to browse?
- Figuring out how to use GitHub may intimidate people
- Programmers occasionally fork the repo

I lean towards staying on GitHub, but I know how to use GitHub, 

For a non-self-hosted solution, itch.io might be one place to look at, at least for submissions.

**Comment by mcwhittemore** [comment]
I recently found out about NaNoGenMo. This repo has been really cool to read. I agree that more instruction could be put in the README. It also might be wise to have a single repo with labels for each year rather than many repos.

Per the github vs other service question, github is already a part of my normal workflow. This default behavior means I'm much more likely to keep up with what is going on here than if it's in another place. Of course, this is just me, there might be another tool that is already part of people's daily lives. That said, I'd say that if a service/tool is new (ie, people aren't using it yet) it's probs not better than a tool that people in this community are using if only because it means that people have to add it to their daily flow.

**Comment by bcj** [comment]
I would probably vote for keeping with the current system. It seems like the benefits/effort ratio is good. I guess I don't know if I'd feel the same if I wasn't a programmer who already had a github account though...

**Comment by greg-kennedy** [comment]
After some thought, I think Github is fine.  The biggest issue is finding out where the "action" is.  So...
- Rename https://github.com/dariusk/NaNoGenMo to NaNoGenMo-2013 so it doesn't seem like the perpetual repo
- Add a bunch more
  # LARGE TEXT
  to readme.md pointing people to Issues tab

**Comment by ikarth** [comment]
Of course, if the goal is to find out if new people with no GitHub experience would find another platform more approachable, we'll have a limited window on that via a GitHub discussion...

**Comment by adregan** [comment]
It would be cool to have a "github pages" page with a list of links to the projects (just so it's easier to see them all). I hesitate to suggest that you add your project link via PR as that might not be very accessible for all.

**Comment by ikarth** [comment]
That's a good idea. It wouldn't be too hard to designate someone to do the additions at the same time the completed tags get added, so those unfamiliar with pull requests don't need to mess with them...

**Comment by MichaelPaulukonis** [comment]
Has THIS been pointed out?

https://github.com/nanogenmo

having the repo's under an org might be a good idea, if we're going with github (And if one of the pre-reqs is "post the code online" that feeds into it, and still doesn't remove the "even though you don't have to use github".

Putting a code-based project on Reddit seems... weird. Plus, problematic for people who don't have reddit accounts, and still need a place to post code and for people who break out into hives when they get near Reddit.

Wherever we go - participants still have to create an account and figure out how to post.
So wherever we go, there should be a step-by-step guide (carefully vetted by non-programmers w/o an account on the target) to signing up.

**Comment by MichaelPaulukonis** [comment]
So.... a little more than a week away

How about we continue with github this year, but put it under the NaNoGenMor org?

We can continue to discuss alternatives throughout the next year, with a goal of reaching a decision prior to October 2017, leaving a month for promotion of the chosen location?

**Comment by greg-kennedy** [comment]
Can we get the 2016 repo up?  I already have resources to share.

**Comment by hugovk** [comment]
If we do put it on https://github.com/NaNoGenMo should it be:

https://github.com/NaNoGenMo/NaNoGenMo-2015

Or:

https://github.com/NaNoGenMo/2015

?

**Comment by mcwhittemore** [comment]
I'd vote for https://github.com/NaNoGenMo/2016

**Comment by dariusk** [comment]
I'll get this going.

**Comment by dariusk** [comment]
It is happening: https://github.com/NaNoGenMo/2016/issues

**Comment by MichaelPaulukonis** [comment]
Thanks Darius - I tried to get to this over lunch... but no DNS for me!

**Comment by enkiv2** [comment]
I agree with Michael and Greg -- while it's important to consider whether
to continue with github, it's probably too late to make this decision for
_this_ year, since by this point in previous years it was typical for
people to already be posting plans or resources.

**Comment by enkiv2** [comment]
Neither, since it's 2016 ;-)

Does github allow project names that begin with a number or contain only
numbers? I'd go with NaNoGenMo-2016 only because if I do a clone for some
reason I don't really want a directory called "2016".

**Comment by greg-kennedy** [comment]
ron_paul_lasers.gif

**Comment by hugovk** [comment]
@enkiv2 https://github.com/NaNoGenMo/2016 is up! If you do fork it, you can go into settings and change the repo name to NaNoGenMo-2016 or something:

![image](https://cloud.githubusercontent.com/assets/1324225/19700620/af445376-9b00-11e6-9ba4-0469930e2fb2.png)

**Comment by agladysh** [comment]
NB: When I search for `NaNoGenMo` and even for `NaNoGenMo 2016`, 2016 site and repo are not in the first results, which is rather frustrating. Is there everything OK with robots.txt etc. there?

**Comment by hugovk** [comment]
I see https://github.com/NaNoGenMo/2016 as the second result for "NaNoGenMo 2016". And the previous year's repos are all linking forward to 2016's. I doubt we have any control over robots.txt etc. on GitHub.

**Comment by agladysh** [comment]
I now see it as a third result. Weird. Ok, sorry for the noise then.

**Comment by dranorter** [comment]
So, I realize I'm way early here, but, is github what's going to be happening for 2017? I see there's some call above for 'discussion throughout the year'.

**Comment by hugovk** [comment]
It'll probably be on GitHub, but not necessarily! In any case, the 2016 repo will be updated to point to wherever it happens closer to the time. 

There's some discussion at this group: https://groups.google.com/forum/m/#!forum/generativetext

And the #nanogenmo channel of the #botALLY Slack. https://botally.slack.com
